### PR TITLE
[FrameworkBundle] Display a proper warning for deprecated --env and --no-debug options

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 4.2
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Currently hidden (remember https://github.com/symfony/symfony/pull/23328)

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Console/Application.php [hunk 1]
```diff
@@ -62,12 +62,19 @@ public function getKernel()
...
+            $notice = 'The "--env" option and its "-e" shortcut are deprecated since Symfony 4.2. Set the "APP_ENV" environment variable instead.';
+            $io = new SymfonyStyle($input, $output instanceof ConsoleOutputInterface ? $output->getErrorOutput() : $output);
+            $io->warning($notice);
+            @trigger_error($notice, E_USER_DEPRECATED);
         }
 
         if ($input->hasParameterOption('--no-debug', true)) {
-            @trigger_error('The "--no-debug" option is deprecated since Symfony 4.2. Set the "APP_DEBUG" environment variable to "0" instead.', E_USER_DEPRECATED);
+            $notice = 'The "--no-debug" option is deprecated since Symfony 4.2. Set the "APP_DEBUG" environment variable to "0" instead.';
+            ($io ?: new SymfonyStyle($input, $output instanceof ConsoleOutputInterface ? $output->getErrorOutput() : $output))->warning($notice);
```
Comment (ogizanagi) [inline src/Symfony/Bundle/FrameworkBundle/Console/Application.php]: Not much relevant, but:

```diff
- $io ?: new SymfonyStyle
+ $io ?? new SymfonyStyle
```

File: src/Symfony/Bundle/FrameworkBundle/Console/Application.php [hunk 2]
```diff
@@ -62,12 +62,19 @@ public function getKernel()
      */
     public function doRun(InputInterface $input, OutputInterface $output)
     {
+        $io = null;
+
         if ($input->hasParameterOption(array('-e', '--env'), true)) {
-            @trigger_error('The "--env" option and its "-e" shortcut are deprecated since Symfony 4.2. Set the "APP_ENV" environment variable instead.', E_USER_DEPRECATED);
+            $notice = 'The "--env" option and its "-e" shortcut are deprecated since Symfony 4.2. Set the "APP_ENV" environment variable instead.';
+            $io = new SymfonyStyle($input, $output instanceof ConsoleOutputInterface ? $output->getErrorOutput() : $output);
```
Comment (ogizanagi) [inline src/Symfony/Bundle/FrameworkBundle/Console/Application.php]: What about using `->getErrorStyle()` instead?

**Review by ogizanagi — APPROVED** [review]

**Review by Tobion — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Console/Application.php [hunk 3]
```diff
@@ -63,11 +63,16 @@ public function getKernel()
     public function doRun(InputInterface $input, OutputInterface $output)
     {
         if ($input->hasParameterOption(array('-e', '--env'), true)) {
-            @trigger_error('The "--env" option and its "-e" shortcut are deprecated since Symfony 4.2. Set the "APP_ENV" environment variable instead.', E_USER_DEPRECATED);
+            $notice = 'The "--env" option and its "-e" shortcut are deprecated since Symfony 4.2. Set the "APP_ENV" environment variable instead.';
```
[the remark this answers is not in the document]
Comment (chalasr) [reply src/Symfony/Bundle/FrameworkBundle/Console/Application.php]: done

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**